rust/lance-index/src/scalar/json.rs

```diff
@@ -832,6 +834,19 @@ impl ScalarIndexPlugin for JsonIndexPlugin {
             .await?;
         Ok(Arc::new(JsonIndex::new(target_index, json_details.path)))
     }
+
+    fn details_as_json(&self, details: &prost_types::Any) -> Result<String> {
+        let registry = self.registry().unwrap();
+        let json_details = crate::pb::JsonIndexDetails::decode(details.value.as_slice())?;
+        let target_details = json_details.target_details.as_ref().expect_ok()?;
+        let target_plugin = registry.get_plugin_by_details(target_details).unwrap();
+        let target_details_json = target_plugin.details_as_json(target_details)?;
+        Ok(serde_json::json!({
+            "path": json_details.path,
+            "target_details": target_details_json,
+        })
```

Comment:
The nested details means that just one `json.loads()` later won't be enough, which is annoying.

Reply:
Done